utility/common.go

```diff
@@ -13,3 +13,27 @@ func CleanUrlPath(p string) string {
 	url.Path = path.Clean(url.Path)
 	return url.String()
 }
+
+// InsertInMap modifies `data` *in place*, inserting `values` at the given `key` or at the top level if the `key`is "".
+// If `key` doesn't exist, it gets created.
+// If `key` exists and its value is not a map, InsertInMap does nothing.
+func InsertInMap(data map[string]interface{}, key string, values map[string]interface{}) {
+	if data == nil || values == nil {
+		return
+	}
```

Comment:
Why not create the map here in case it is nil?

Reply:
Because i am not `return`ing data, if i create the map there i wouldn't be able to "see it" from the outside as it would a be (new) local reference.
I could `return` it but i'm very much against mutating _and_ returning data, is very hacky and counter intuitive. 
I could also make a full copy of the map and return that, but that is expensive as it is used at the payload level.